pageserver/src/tenant/layer_map.rs

```diff
@@ -499,87 +263,226 @@ where
         key_range: &Range<Key>,
         lsn: Lsn,
     ) -> Result<Vec<(Range<Key>, Option<Arc<L>>)>> {
-        let mut points = vec![key_range.start];
-        let envelope = AABB::from_corners(
-            [IntKey::from(key_range.start.to_i128()), IntKey::from(0)],
-            [
-                IntKey::from(key_range.end.to_i128()),
-                IntKey::from(lsn.0 as i128),
-            ],
-        );
-        for e in self
-            .historic_layers
-            .locate_in_envelope_intersecting(&envelope)
-        {
-            let l = &e.layer;
-            assert!(l.get_lsn_range().start <= lsn);
-            let range = l.get_key_range();
-            if key_range.contains(&range.start) {
-                points.push(l.get_key_range().start);
-            }
-            if key_range.contains(&range.end) {
-                points.push(l.get_key_range().end);
-            }
+        // TODO I'm 80% sure the lsn bound is inclusive. Double-check that
+        //      and document it. Do the same for image_layer_exists and count_deltas
```

Comment:
@hlinnaka It's not that simple to infer from the caller code. Is the bound supposed to be inclusive or exclusive? I'd expect it to be exclusive because that's how it is on `search`. But when I implement it as exclusive I get duplicate image layers, so I'm guessing it's not.

Same question for `image_layer_exists` and `count_deltas` if you happen to know the answer.

Reply:
It's inclusive. I'll update the docs and make `count_deltas` treat the bound as inclusive